packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -114,26 +121,16 @@ export class GatewayRoute extends cdk.Resource implements IGatewayRoute {
 /**
  * Interface with properties necessary to import a reusable GatewayRoute
  */
-interface GatewayRouteAttributes {
+export interface GatewayRouteAttributes {
   /**
    * The name of the GatewayRoute
    */
-  readonly gatewayRouteName?: string;
-
-  /**
-   * The Amazon Resource Name (ARN) for the GatewayRoute
-   */
-  readonly gatewayRouteArn?: string;
+  readonly gatewayRouteName: string;
 
   /**
-   * The name of the mesh this GatewayRoute is associated with
+   * The name of the VirtualGateway this GatewayRoute is associated with
    */
-  readonly meshName?: string;
-
-  /**
-   * The name of the Virtual Gateway this GatewayRoute is associated with
-   */
-  readonly virtualGateway?: IVirtualGateway;
+  readonly virtualGateway: IVirtualGateway;
```

Comment:
In EKS, for example we actually only take ARNs here and not strongly-typed objects.

Reply:
@skinny85 asked for these to be strongly typed. An ARN would work for this, but this using the constructs is more explicit.

Reply:
Yes, we always use the strongly-typed properties for these, surprised EKS does something different.